ext/nnstreamer/tensor_filter/tensor_filter_cpp.cc

```diff
@@ -200,16 +204,19 @@ int tensor_filter_cpp::open (const GstTensorFilterProperties *prop, void **priva
       return -EINVAL; /** Model file / name not found */
     }
 
-    /** At this stage, the constructor of the .so file should have registered
-      * itself with tensor_filter_cpp_register() */
-    dlclose (handle);
-
     if (filters.find(prop->model_files[0]) == filters.end()) {
       /** It's still not found. it's not there. */
+      dlclose (handle);
       g_printerr_once (__FILE__, __LINE__, "C++ custom filter %s is not found in %s.\n",
           prop->model_files[0], prop->model_files[1]);
       return -EINVAL;
     }
+
+    /** We do not know until when this handle might be required: user may
+      * invoke functions from it at anytime while the pipeline is not
+      * closed */
+    handles.push_back (handle);
```

Comment:
do we need locking around this? (as handle is class static variable shared by all instance of this class).